### ISSUE TITLE
No quick fix for instance_member_access_from_static

### ISSUE THREAD

**Issue by bernaferrari** [dart-lang/sdk#49871]
I get `instance_member_access_from_static`  when... well, doing things wrong.

I believe it would be nice to have a quick fix that made `method1` static:

```dart
class Hello {
  void method1() {}

  static void method2() {
    method1(); // Instance members can't be accessed from a static method
  }
```

**Comment by bwilkerson** [comment]
Another possible fix would be to make `method2` an instance method.

**Comment by bernaferrari** [comment]
Agreed, but from my experience when I make 1 static, everything screams. Having something static when it doesn't need to be is a lot more rare FOR ME.

**Comment by bwilkerson** [comment]
To be clear, I believe that we should have both, not one or the other. My comment was in no way intended to imply that I thought your suggestion wasn't both reasonable and valid.

**Comment by bwilkerson** [comment]
Your CL raises an interesting question, and I thought it would be better to include more people in the discussion.

Under what conditions should we offer to make a method (or other instance member) static? Doing so will result in more diagnostics in at least two cases:
* the member being converted references instance members on `this` other than itself, and
* there are references to the member that use an instance of the class to access it.

(There might be others that I'm not thinking of at the moment.)

We can't take the time required to check for the second condition and still be responsive, but we could check for the first condition in a reasonable amount of time. Should we? For example, should we offer the fix in the following case:

```
class C {
  int x;
  int y;

 C(this.x, this.y);

  int m1() => x * y;

  static int m2() => product() / 2;
}
```

It seems unlikely that a user would want to convert a method that references instance fields to be `static`.

Even if you agree that we shouldn't offer the fix in that case, what if the instance member referenced in `m1` was another method and not a field (or, more accurately, getter)? Where do we draw the line?
